### PR TITLE
fix: Added animation support for visible prop

### PR DESCRIPTION
**Pull request by tleonh**
The issue affects Banner component

Problem:
- When you activate the "visible" prop, It won´t do anything. It should appear or disappear with an animation.

Fix:
- Pass a function to save status to updaters startCollapse and startExpand on updaters.js.
- Added usePreviousStatus hook to save previous status state and follow component logic.
- Pass a function to save status to updaters switch on BannerAnimator.js.
- Added new validation to collapse Banner after the visible is disabled.

### CODE REVIEW

**Review by danllach — APPROVED** [review]
LGTM

**Review by t-rodrjua — APPROVED** [review]
Nice work.

File: packages/banner/src/BannerAnimator/BannerAnimator.js [hunk 1]
```diff
@@ -105,7 +105,7 @@ const BannerAnimator = props => {
   useEffect(
     () => {
       const { isVisible } = props;
-
+
```
Comment (DevFoncy) [inline packages/banner/src/BannerAnimator/BannerAnimator.js]: Remove this space

File: packages/banner/src/BannerAnimator/BannerAnimator.js [hunk 2]
```diff
@@ -128,23 +128,35 @@ const BannerAnimator = props => {
     [props]
   );
 
+  function usePreviousStatus(value){
```
Comment (DevFoncy) [inline packages/banner/src/BannerAnimator/BannerAnimator.js]: Space between  paramether and the { 
`  function usePreviousStatus(value) {
`

File: packages/banner/src/BannerAnimator/BannerAnimator.js [hunk 3]
```diff
@@ -128,23 +128,35 @@ const BannerAnimator = props => {
...
+    });
+    return ref.current;
+  }
+
+  const prevStatus = usePreviousStatus(status);
+
   useEffect(
     () => {
-      const { status: prevStatus } = props;
+      const expandStatuses = prevStatus === statuses.COLLAPSED && status === statuses.EXPANDING ||
```
Comment (DevFoncy) [inline packages/banner/src/BannerAnimator/BannerAnimator.js]: Replace all the condition using this : 
` const expandStatuses =
        (prevStatus === statuses.COLLAPSED && status === statuses.EXPANDING) ||
        (prevStatus === statuses.COLLAPSING && status === statuses.EXPANDING);
`

File: packages/banner/src/BannerAnimator/BannerAnimator.js [hunk 4]
```diff
@@ -128,23 +128,35 @@ const BannerAnimator = props => {
...
       }
-      if (prevStatus === statuses.COLLAPSING && status === statuses.EXPANDING) {
+      if (expandStatuses) {
         expand();
         return;
       }
       if (prevStatus === statuses.EXPANDING && status === statuses.COLLAPSING) {
         collapse();
       }
+
```
Comment (DevFoncy) [inline packages/banner/src/BannerAnimator/BannerAnimator.js]: Remove. this space blank

File: packages/banner/src/BannerAnimator/BannerAnimator.js [hunk 5]
```diff
@@ -128,23 +128,35 @@ const BannerAnimator = props => {
...
         expand();
         return;
       }
       if (prevStatus === statuses.EXPANDING && status === statuses.COLLAPSING) {
         collapse();
       }
+     
     },
-    [props]
+    [props,prevStatus, status]
```
Comment (DevFoncy) [inline packages/banner/src/BannerAnimator/BannerAnimator.js]: `    [props, prevStatus, status]
`

**Review by DevFoncy — CHANGES_REQUESTED** [review]
Dont forget enable eslint config

**Comment by tleonh** [comment]
@DevFoncy applied eslint changes

**Pull request merged**